### PR TITLE
Add Blacklist Functionality to PreferencePage

### PR DESCRIPTION
**Pull request by js1998**

### CODE REVIEW

File: imports/redux/reducers/index.jsx [hunk 1]
```diff
@@ -94,21 +68,31 @@ function userReducer(state = initialUserState, action) {
...
       break; //TODO: remove after reducer refactor
     case LOGOUT_USER:
-      return { ...state, isSignedIn: false, fullName: "", email: "", blacklist: [], favourites: [] };
+      return { ...state, isSignedIn: false, fullName: "", email: "", userId: "", blacklist: [], favourites: [] };
     case ADD_BLACKLIST:
-      let addedBlacklist = state.blacklist.slice();
-      addedBlacklist.push(action.blacklist);
-      return { ...state, blacklist: addedBlacklist };
+      let matchedUsers = UserInfo.update({_id: state.userId}, { $push:{ "preferences.blacklist": action.blacklist } })
+      if(matchedUsers === 0){
```
Comment (hailin-zhang) [inline imports/redux/reducers/index.jsx]: space

File: imports/redux/reducers/index.jsx [hunk 2]
```diff
@@ -94,21 +68,31 @@ function userReducer(state = initialUserState, action) {
...
+        console.log("UserId Does Not Exist in Database")
```
Comment (hailin-zhang) [inline imports/redux/reducers/index.jsx]: can you make a note to remove console logs

File: imports/redux/reducers/index.jsx [hunk 3]
```diff
@@ -94,21 +68,31 @@ function userReducer(state = initialUserState, action) {
...
       let updatedBlacklist = Array.filter((value, index, array) => {
         return action.blacklistToRemove !== value;
       })
       return { ...state, blacklist: updatedBlacklist };
     case ADD_FAVOURITES:
-      let addedFavourites = state.favourites.slice();
-      addedFavourites.push(action.favourite);
-      return { ...state, favourites: addedFavourites };
+      matchedUsers = UserInfo.update({_id: state.userId}, { $push:{ "preferences.favourites": action.favourite } })
+      if(matchedUsers === 0){
```
Comment (hailin-zhang) [inline imports/redux/reducers/index.jsx]: space

File: imports/redux/reducers/index.jsx (hunk 2 above)
Comment (js1998) [reply imports/redux/reducers/index.jsx]: Done :)

**Pull request merged**